src/lib/steps/ngc.ts

```diff
@@ -1,49 +1,122 @@
+  options.skipTemplateCodegen = true;
+  options.strictMetadataEmit = true;
+  options.flatModuleId = ngPkg.fullPackageName;
+  options.flatModuleOutFile = `${ngPkg.flatModuleFileName}.js`;
 
-  if (ngPkg.jsxConfig) {
-    debug('prepareTsConfig: Applying jsx flag to tsconfig ' + ngPkg.jsxConfig);
-    tsConfig['compilerOptions']['jsx'] = ngPkg.jsxConfig;
+  // invoke ngc programmatic API
+  let emitFlags = ng.EmitFlags.Default;
+  if (!(options.skipMetadataEmit || options.flatModuleOutFile)) {
+      emitFlags |= ng.EmitFlags.Metadata;
+  }
+  if (options.skipTemplateCodegen) {
+      emitFlags = emitFlags & ~ng.EmitFlags.Codegen;
   }
 
-  await writeJson(outFile, tsConfig);
+  return {
+    rootNames: [ ngPkg.entryFile ],
+    options: options,
+    emitFlags: emitFlags
+  };
 }
 
-
 /**
  * Compiles typescript sources with 'ngc'.
  *
  * @param ngPkg Angular package data
  * @param basePath
+ * @returns Promise<string> Path of the flatModuleOutFile
  */
 export async function ngc(ngPkg: NgPackageData, basePath: string): Promise<string> {
-  const tsConfigPath: string = `${basePath}/tsconfig.lib.json`;
-  debug(`ngc ${tsConfigPath}, { basePath: ${basePath} })`);
+  debug(`ngc`);
 
-  await prepareTsConfig(ngPkg, tsConfigPath);
+  // const opts = createOptions(ngPkg, basePath);
 
-  // invoke ngc programmatic API
-  const compilerConfig = readConfiguration(tsConfigPath);
-  performCompilation(compilerConfig);
+  // Read the default configuration and overwrite package-specific options
+  const tsConfig = ng.readConfiguration(path.resolve(__dirname, '..', 'conf', 'tsconfig.ngc.json'));
+  tsConfig.rootNames = [ path.resolve(basePath, ngPkg.entryFile) ];
+  tsConfig.options.flatModuleId = ngPkg.fullPackageName
+  tsConfig.options.flatModuleOutFile = `${ngPkg.flatModuleFileName}.js`;
+  tsConfig.options.basePath = basePath;
+  tsConfig.options.baseUrl = basePath;
+  tsConfig.options.outDir = path.resolve(basePath, '.ng_pkg_build');
+  tsConfig.options.genDir = path.resolve(basePath, '.ng_pkg_build');
+
+  /*
+  const compilerHost = ng.createCompilerHost({ options: tsConfig.options });
+  const program = ng.createProgram({
+    rootNames: tsConfig.rootNames,
+    options: tsConfig.options,
+    host: compilerHost
+  });
+
+  const result = ng.performCompilation({
+    options: tsConfig.options,
+    rootNames: tsConfig.rootNames,
+    emitFlags: tsConfig.emitFlags,
+    // oldProgram: program
+  });
+  console.log(result);
+  */
+
+
+  // --> XX: start custom transformer
+  // Hook into TypeScript transformation API
+  const customTransformers: ng.CustomTransformers = {
+    beforeTs: [ componentTransformer ]
+  };
 
-  debug('Reading tsconfig from ' + tsConfigPath);
-  const tsConfig = await readJson(tsConfigPath);
+  // Invoke ngc programmatic API
+  const result = ng.performCompilation({
+    options: tsConfig.options,
+    rootNames: tsConfig.rootNames,
+    emitFlags: tsConfig.emitFlags,
+    customTransformers
```

Comment:
Passing `customTransformers` seems not to work.

AoT metadata still includes `templateUrl`, but the transpiled JavaScript includes the transformation.

```json
"arguments":[{"selector":"baz-component","templateUrl":"./baz.component.html", ... }
```

```js
import { Component } from '@angular/core';
"foobar!"
export class BazComponent {
}
//# sourceMappingURL=baz.component.js.map
```

Reply:
Solved! No we're talking!